keps/sig-instrumentation/0034-distributed-tracing-kep.md

```diff
@@ -0,0 +1,299 @@
+
+GA
+
+- [] Versioning for span naming and backwards-compatibility guarantees
+
+## Production Readiness Survey
+
+* Feature enablement and rollback
+  - How can this feature be enabled / disabled in a live cluster?  **Feature-gate**
+  - Can the feature be disabled once it has been enabled (i.e., can we roll
+    back the enablement)?  **Yes, the feature gate can be disabled in all relevant components**
+  - Will enabling / disabling the feature require downtime for the control
+    plane?  **Yes, control-plane components must be restarted with the feature-gate disabled.**
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node?  **No, it just requires restarting the kubelet with the feature-gate disabled**
+  - What happens if a cluster with this feature enabled is rolled back? **No Adverse effects** What
+    happens if it is subsequently upgraded again?  **No adverse effects**
+  - Are there tests for this?  **No.  The feature hasn't been developed yet.**
+* Scalability
+  - Will enabling / using the feature result in any new API calls? **No (there was a recent change in the KEP to that effect).**
+    Describe them with their impact keeping in mind the [supported limits][]
+    (e.g. 5000 nodes per cluster, 100 pods/s churn) focusing mostly on:
+     - components listing and/or watching resources they didn't before
+     - API calls that may be triggered by changes of some Kubernetes
+       resources (e.g. update object X based on changes of object Y)
+     - periodic API calls to reconcile state (e.g. periodic fetching state,
+       heartbeats, leader election, etc.)
+  - Will enabling / using the feature result in supporting new API types? **No**
+    How many objects of that type will be supported (and how that translates
+    to limitations for users)?
+  - Will enabling / using the feature result in increasing size or count
+    of the existing API objects?  **Yes.  It adds an annotation to traced objects.**
+  - Will enabling / using the feature result in increasing time taken
+    by any operations covered by [existing SLIs/SLOs][] (e.g. by adding
+    additional work, introducing new steps in between, etc.)? **No**
+    Please describe the details if so.
+  - Will enabling / using the feature result in non-negligible increase
+    of resource usage (CPU, RAM, disk IO, ...) in any components?
+    Things to keep in mind include: additional in-memory state, additional
+    non-trivial computations, excessive access to disks (including increased
+    log volume), significant amount of data sent and/or received over
+    network, etc. Think through this in both small and large cases, again
+    with respect to the [supported limits][].  **The tracing client library has an in-memory cache for outgoing spans.  I believe this is limited to 1Mb by default.  This overhead would apply to all controllers that export spans.**
```

Comment:
This includes, amongst others, Kubelet. I think in the context of this question, mentioning node components may be more important.

Reply:
I added a note that this applies to the kubelet as well.